### ISSUE TITLE
AWS S3 Config file missing secret key

### ISSUE THREAD

**Issue by SleepyBandit** [cat-box/aws-foundry-ssl#4]
I went through the setup and it was smooth and easy to setup. However when I attempted to use the S3 bucket in-world to upload content it failed to upload. I checked the AWS.json file in the config folder on the EC2 instance and the secretAccessKey value was not set. It looked to still be set to a default value.

I went into iams and generated a new accesskeyId and secretAccessKey pair for the existing user account, (confirmed it was the right one by matching the existing accesskeyId in the file) and updated the AWS.json file wiith the new pair value. However, even after doing all this and restarting the instance, it still will not work with S3. It writes the following error.
![image](https://user-images.githubusercontent.com/6467347/100667797-d2da9b00-330f-11eb-99b1-c8a0af3879e3.png)

What other areas need to be changed to make this work appropriately?

**Comment by zkkng** [comment]
Can you pull the User Data log from your instance? All the commands work and I have tested the bucket previously so my top suspicions are either some update broke the functionality which is not likely, or it is a race condition.

It could be the implicit dependency is not triggered since the GetAtt is in the User Data join block.

**Comment by SleepyBandit** [comment]
> Can you pull the User Data log from your instance and make sure the command to get the userkey did not error out? All the commands work and I have tested the bucket previously so my top suspicions are either some update broke the functionality which is not likely, or it is a race condition.
> 
> It could be the implicit dependency on UserKey is not triggered since the GetAtt is in the User Data join block.

Are the logs under /foundrydata/logs?
The only log in the error.log file is related to chat "Required field content not present in ChatMessage" otherwise its empty.
The debug.log file has details about my attempts to reconfigure AWS but it says it was successful.
{"message":"Configured AWS credentials using /foundrydata/Config/AWS.json","level":"info","timestamp":"2020-11-30 21:34:14"}

**Comment by SleepyBandit** [comment]
I think I got it to work...
I had generated the new accesskeyId and secretAccessKey pair like I mentioned before, but had left the old key pair active on the user since it typically doesn't hurt anything to have two key pairs generated. After going in and deactivating the old key pair, then restarting the foundry instance, it appears to be connecting. So in this situation it is important to deactivate / remove the old pair if switching.

Still, for anyone else who followed instructions and S3 integration doesn't work out-of-the-box, check your AWS.json file via SSH or SFTP and see if the SecretAccessKey is set properly. In my case it was not, was still set to a default value, and required me to take the extra steps I outlined above.
